File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/bot/BotRuntimeSessionResponse.java
package com.mieai.qqbot.admin.bot;

import com.mieai.qqbot.runtime.supervisor.BotSessionSnapshot;
import java.util.Objects;

public record BotRuntimeSessionResponse(String id, long sequence) {

    static BotRuntimeSessionResponse from(BotSessionSnapshot snapshot) {
        Objects.requireNonNull(snapshot, "snapshot must not be null");
        return new BotRuntimeSessionResponse(snapshot.sessionId(), snapshot.sequence());
    }
}

File: qqbot-module-qqbot-runtime/src/main/kotlin/com/mieai/qqbot/admin/bot/BotRuntimeSessionResponse.kt
package com.mieai.qqbot.admin.bot

import com.mieai.qqbot.runtime.supervisor.BotSessionSnapshot

data class BotRuntimeSessionResponse(val id: String, val sequence: Long) {
    companion object {
        fun from(snapshot: BotSessionSnapshot): BotRuntimeSessionResponse =
            BotRuntimeSessionResponse(snapshot.sessionId, snapshot.sequence)
    }
}
